### DISCUSSION TITLE
I want to use ref inside a function.

### DISCUSSION THREAD

**Discussion by cyb9701** [rrousselGit/riverpod#2558] [Q&A]
I want to use ref inside a function.
I don't know how to use ref in a function, so I receive WidfetRef as a parameter and use that ref.
Is this the only way to use ref inside a function?

```dart
class CustomRouter {
  static Future<void> routing(WidgetRef ref) async {
    // Use ref here
  }
}
```

**Comment by mafreud** [comment]
With (Async)Notifier, you can use ref everywhere inside the class.
https://docs-v2.riverpod.dev/docs/providers/notifier_provider
